runtime/vm/OutOfLineINL_java_lang_invoke_NativeMethodHandle.cpp

```diff
@@ -199,6 +207,119 @@ OutOfLineINL_java_lang_invoke_NativeMethodHandle_freeJ9NativeCalloutDataRef(J9VM
 	VM_OutOfLineINL_Helpers::returnVoid(currentThread, 1);
 	return EXECUTE_BYTECODE;
 }
-#endif /* J9VM_OPT_PANAMA */
 
+/* jdk.internal.nicl.NativeInvoker:
+ * private final static native long generateNativeFunctionPointerBinding(
+ * 																			Class<?> returnType,
+ * 																			Class<?>[] parameterTypes,
+ * 																			Class<? extends Object> anonClass,
+ * 																			Object anonInstance,
+ * 																			String methodSignature
+ * 																		)
+ */
+
+VM_BytecodeAction
+OutOfLineINL_jdk_internal_nicl_NativeInvoker_generateNativeFunctionPointerBinding(J9VMThread *currentThread, J9Method *method)
+{
+	J9JavaVM *vm = currentThread->javaVM;
+	j9object_t methodSignatureObj = *(j9object_t*)currentThread->sp;
+	jobject anonInstance = (jobject)(currentThread->sp + 1);
+	jclass anonClassRef = (jclass)(currentThread->sp + 2);
+	j9object_t paramTypeArrayObj = *(j9object_t*)(currentThread->sp + 3);
+	j9object_t returnTypeObj = *(j9object_t*)(currentThread->sp + 4);
+	J9InternalVMFunctions* vmFuncs = vm->internalVMFunctions;
+	VM_BytecodeAction rc = EXECUTE_BYTECODE;
+	I_64 functionAddress = 0;
+	char buf[SIGNATURE_LENGTH];
+	char *signature = buf;
+	U_16 length = 0;
+	jmethodID methodID = NULL;
+	J9Class *typeBuf[16];
+	J9Class **paramTypes = typeBuf;
+	U_32 paramSize = 0;
+	JITBuilderReturnCode retCode = NO_ERROR;
+
+	VM_OutOfLineINL_Helpers::buildInternalNativeStackFrame(currentThread, method);
+
+	length = (U_16)vmFuncs->getStringUTF8Length(currentThread, methodSignatureObj);
+	if (length > SIGNATURE_LENGTH) {
+		PORT_ACCESS_FROM_VMC(currentThread);
+		signature = (char *) j9mem_allocate_memory(sizeof(char) * (length + 1) , OMRMEM_CATEGORY_VM);
+		if (NULL == signature) {
+			rc = GOTO_THROW_CURRENT_EXCEPTION;
+			setNativeOutOfMemoryError(currentThread, 0, 0);
+			goto popMethodTypeFrame;
+		}
+	}
+
+	if (UDATA_MAX == vmFuncs->copyStringToUTF8Helper(currentThread, methodSignatureObj, TRUE, J9_STR_NONE, (U_8*) signature, length + 1)) {
+		rc = GOTO_THROW_CURRENT_EXCEPTION;
+		setCurrentException(currentThread, J9VMCONSTANTPOOL_JAVALANGINTERNALERROR, NULL);
+		goto popMethodTypeFrame;
+	}
+
+	methodID = (jmethodID) getMethodOrFieldID((JNIEnv *) currentThread, anonClassRef, "fn", signature, J9JNIID_METHOD, FALSE);
```

Comment:
If the Java code looks up the reflect method and passes it to the function, the native code can use `FromReflectedMethod` to get the methodID.  Really simplifies the native code